monitoring/garden_collector.go

```diff
@@ -0,0 +1,120 @@
+package monitoring
+
+import (
+	"sync"
+
+	"github.com/lightninglabs/taproot-assets/tapgarden"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// assetBalancesCollector is a Prometheus collector that exports the balances
+// of all taproot assets.
+type gardenCollector struct {
+	collectMx sync.Mutex
+
+	cfg      *PrometheusConfig
+	registry *prometheus.Registry
+
+	pendingBatches   *prometheus.GaugeVec
+	completedBatches prometheus.Gauge
+}
+
+func newGardenCollector(cfg *PrometheusConfig,
+	registry *prometheus.Registry) *gardenCollector {
+
+	return &gardenCollector{
+		cfg:      cfg,
+		registry: registry,
+		pendingBatches: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "mint_batches",
+				Help: "Batched mint transactions",
+			},
+			[]string{"batch_pubkey"},
+		),
+		completedBatches: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Name: "completed_batches",
+				Help: "Total number of completed mint batches",
+			}),
+	}
+}
+
+// Describe sends the super-set of all possible descriptors of metrics
+// collected by this Collector to the provided channel and returns once the
+// last descriptor has been sent.
+//
+// NOTE: Part of the prometheus.Collector interface.
+func (a *gardenCollector) Describe(ch chan<- *prometheus.Desc) {
+	a.collectMx.Lock()
+	defer a.collectMx.Unlock()
+
+	a.pendingBatches.Describe(ch)
+	a.completedBatches.Describe(ch)
+}
+
+// Collect is called by the Prometheus registry when collecting metrics.
+//
+// NOTE: Part of the prometheus.Collector interface.
+func (a *gardenCollector) Collect(ch chan<- prometheus.Metric) {
+	a.collectMx.Lock()
+	defer a.collectMx.Unlock()
+
+	if a.cfg == nil {
```

Comment:
These checks could be done in `newGardenCollector` too, otherwise it's easy to just deploy something that'll not collect (but rather just error log).

Reply:
Also did this for the rest of the collectors